src/steps/UploadFlow.tsx

```diff
@@ -110,7 +123,8 @@ export const UploadFlow = ({ nextStep }: Props) => {
         <MatchColumnsStep
           data={state.data}
           headerValues={state.headerValues}
-          onContinue={(data) => {
+          onContinue={async (values) => {
+            const data = await validationStepHook(values)
```

Comment:
`validationStepHook` should probably be called `matchColumnsStepHook` to be consistent with other hooks. Or we should call hooks `afterMatchColumnsStepHook` or `beforeValidationStepHook` so they would be less confusing about when they run.